src/main/resources/lib/guillotine/schema/schema-creation-callbacks/form-intermediate-step.ts

```diff
@@ -1,6 +1,86 @@
 import { CreationCallback } from '../../utils/creation-callback-utils';
 import { insertOriginalContentTypeField } from './common/original-content-type';
+import { runInLocaleContext } from '../../../../lib/localization/locale-context';
+import * as contentLib from '/lib/xp/content';
+import { CONTENT_LOCALE_DEFAULT } from '../../../../lib/constants';
+import { forceArray } from '../../../../lib/utils/array-utils';
+
+type Step = {
+    label: string;
+    nextStep?: {
+        _selected: 'external' | 'next';
+        external?: {
+            formNumber?: string;
+            externalUrl?: string;
+        };
+        next?: {
+            steps: Step[];
+        };
+    };
+};
+
+type ContentData = {
+    steps: Step[];
+};
+
+const updateStepFormNumbers = (step: Step, defaultLayerStep: Step | undefined): Step => {
+    if (!defaultLayerStep) {
+        return step;
+    }
+
+    if (step.nextStep?._selected === 'external' && !step.nextStep.external?.formNumber) {
+        if (
+            defaultLayerStep.nextStep?._selected === 'external' &&
+            defaultLayerStep.nextStep.external?.formNumber
+        ) {
+            if (step.nextStep.external) {
+                step.nextStep.external.formNumber = defaultLayerStep.nextStep.external.formNumber;
+            }
+        }
+    }
+
+    if (step.nextStep?._selected === 'next' && step.nextStep.next?.steps) {
+        if (
+            defaultLayerStep.nextStep?._selected === 'next' &&
+            defaultLayerStep.nextStep.next?.steps
+        ) {
+            const currentSteps = forceArray(step.nextStep.next.steps);
+            const defaultSteps = forceArray(defaultLayerStep.nextStep.next.steps);
+
+            step.nextStep.next.steps = currentSteps.map((currentStep, index) =>
+                updateStepFormNumbers(currentStep, defaultSteps[index])
+            );
+        }
+    }
+
+    return step;
+};
 
 export const formIntermediateStepCallback: CreationCallback = (context, params) => {
     insertOriginalContentTypeField(params);
+
+    params.fields.data.resolve = (env) => {
+        if (env.source.language !== CONTENT_LOCALE_DEFAULT) {
+            const defaultLayerContent = runInLocaleContext(
+                { locale: CONTENT_LOCALE_DEFAULT, branch: 'draft' },
+                () => contentLib.get({ key: env.source._id })
+            );
+
+            if (defaultLayerContent?.data?.steps) {
+                contentLib.modify({
```

Comment:
Callbacks blir kalt hver gang frontend kaller et innhold fra Enonic, så jeg lurer på om vi kanskje ikke bør ha mutering av innhold her, men heller legge det i kontrolleren. Controllers kjører når redaktøren åpner et innhold og så lagrer det.

Du har feks `controllers/form-intermediate-step-controller.ts`. Men om du vil så kan vi la den ligge i callback mens vi tester et "proof of concept" og så flytter vi den over til controlleren?

Reply:
Flyttet til controlleren 👍